Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/DirectPushService.cs

```diff
@@ -36,29 +36,35 @@ public async Task SendNotification(IdentityAddress recipient, object notificatio
             var pnsConnector = _pnsConnectorFactory.CreateFor(platform);
 
             var sendResults = await pnsConnector.Send(group, recipient, notification);
-            await ParseNotificationResponses(sendResults);
-
-            _logger.LogTrace($"Successfully sent push notifications to identity '{recipient}' on platform '{Enum.GetName(platform)}'");
+            await HandleNotificationResponses(sendResults);
         }
     }
 
-    private async Task ParseNotificationResponses(IEnumerable<SendResult> sendResults)
+    private async Task HandleNotificationResponses(SendResults sendResults)
     {
-        foreach (var sendResult in sendResults.Where(sendResult => sendResult.IsFailure))
+        var devicesIdsToDelete = new List<DeviceId>();
+        foreach (var sendResult in sendResults.Failures)
         {
             switch (sendResult.Error.Reason)
             {
-                case SendResult.FailureReason.InvalidHandle:
-                    _logger.LogInformation("Deleting device {deviceId} since handle is no longer valid.", sendResult.Error.DeviceId);
-                    await _registrationRepository.Delete(sendResult.Error.DeviceId, CancellationToken.None);
+                case SendResult.ErrorReason.InvalidHandle:
+                    _logger.LogInformation("Deleting device {deviceId} since handle is no longer valid.", sendResult.DeviceId);
```

Comment:
We don't delete a device!!
```suggestion
                    _logger.LogInformation("Deleting device registration for '{deviceId}' since handle is no longer valid.", sendResult.DeviceId);
```